src/framedata.cpp

```diff
@@ -0,0 +1,52 @@
+#include "framedata.h"
+
+#include "completion.h"
+#include "layout.h"
+
+using std::string;
+
+size_t LayoutAlgorithmCount() {
```

Comment:
I'm seeing some room for improvement around this whole `g_layout_names` thing, but I'm guessing that you purposely left this untouched to minimize the diff for now (and tackle it later on).
Given that this is newly added code, maybe a TODO/FIXME indicating the temporary nature of it would be good. (That is, of course, only if I'm guessing correctly...)

Also: Why TitleCase for a function name?

Reply:
Yes I tried to minimize the diff, and the `g_layout_names` is still used in some command in `layout.cpp` which I did not want to touch.

The TitleCase is because this was originally part of the enum. But I think I can make it camelCase if you prefer.

Reply:
I've renamed it.